rudder-core/src/main/scala/com/normation/rudder/reports/execution/ReportsExecutionRepositoryImpl.scala

```diff
@@ -0,0 +1,167 @@
+/*
+*************************************************************************************
+* Copyright 2013 Normation SAS
+*************************************************************************************
+*
+* This file is part of Rudder.
+*
+* Rudder is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 3 of the License, or
+* (at your option) any later version.
+*
+* In accordance with the terms of section 7 (7. Additional Terms.) of
+* the GNU General Public License version 3, the copyright holders add
+* the following Additional permissions:
+* Notwithstanding to the terms of section 5 (5. Conveying Modified Source
+* Versions) and 6 (6. Conveying Non-Source Forms.) of the GNU General
+* Public License version 3, when you create a Related Module, this
+* Related Module is not considered as a part of the work and may be
+* distributed under the license agreement of your choice.
+* A "Related Module" means a set of sources files including their
+* documentation that, without modification of the Source Code, enables
+* supplementary functions or services in addition to those offered by
+* the Software.
+*
+* Rudder is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Rudder.  If not, see <http://www.gnu.org/licenses/>.
+
+*
+*************************************************************************************
+*/
+
+package com.normation.rudder.reports.execution
+
+
+import com.normation.inventory.domain.NodeId
+import com.normation.rudder.db.DB
+import com.normation.rudder.repository.jdbc.PostgresqlInClause
+
+import net.liftweb.common._
+import com.normation.rudder.db.Doobie
+import scalaz.{Failure => _, _}, Scalaz._
+import doobie.imports._
+import scalaz.concurrent.Task
+
+case class RoReportsExecutionRepositoryImpl (
+    db: Doobie
+  , pgInClause : PostgresqlInClause
+) extends RoReportsExecutionRepository with Loggable {
+
+  import db._
+
+  override def getNodesLastRun(nodeIds: Set[NodeId]): Box[Map[NodeId, Option[AgentRun]]] = {
+    nodeIds.map( _.value).toList.toNel match {
+      case None => Full(Map())
+      case Some(nodes) =>
+
+        //val nodes = pgInClause.in("nodeid", nodeIds.map(_.value))
+        implicit val nodesParam = Param.many(nodes)
+
+        //notice the use of # ({} is forbidden) in place of ${} to interpolate the actual value of in,
+        //not make a sql parameter to the query
```

Comment:
It's a comment that is related to slick, it need to be change. We can't simply interpolate sql"..." queries, because (of course) there is intelligence in them, they are not simple string. But we used to build the "in" clause differently depending of the number of elements in the "in" clause, and just output the resulting string. That can't be done anymore. 
I don't thing "in" is adapted by doobie. So either we rely in plain string queries (see: http://tpolecat.github.io/doobie-0.2.4/15-FAQ.html#how-do-i-turn-an-arbitrary-sql-string-into-a-queryquery0) and continue to use our "in" clause. Else, we stop using it.